docs/references-docs-in-vfs.md

```diff
@@ -0,0 +1,301 @@
+}
+```
+
+#### Response
+
+```http
+HTTP/1.1 204 No Content
+Content-Type: application/vnd.api+json
+```
+
+### GET /files/:type/:doc-id/relationships/references
+
+Returns all the files associated to an album or playlist.
+
+#### Request
+
+```http
+GET /files/io.cozy.playlists/e9308dc2-e2e3-11e6-b685-fb88662613d4/relationships/references HTTP/1.1
+Content-Type: application/vnd.api+json
+Accept: application/vnd.api+json
+```
+
+#### Response
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/vnd.api+json
+```
+
+```json
+{
+  "data": [
+    { "type": "io.cozy.files", "id": "417c4e58-e2e4-11e6-b7dc-2b68ed7b77f4" },
+    { "type": "io.cozy.files", "id": "4504f55c-e2e4-11e6-88f2-d77aeecab549" },
+    { "type": "io.cozy.files", "id": "4587727a-e2e4-11e6-bfe9-ef1be7df7f26" },
+    { "type": "io.cozy.files", "id": "45d591d0-e2e4-11e6-ab9a-ff3b218e31cc" }
+  ]
+}
+```
+
+### POST /files/:type/:doc-id/relationships/references
+
+When creating an album or a playlist, it's tedious to add the references to it
+for each file individually. This route allows to make it in bulk.
+
+#### Request
+
+```http
+POST /files/io.cozy.playlists/e9308dc2-e2e3-11e6-b685-fb88662613d4/relationships/references HTTP/1.1
+Content-Type: application/vnd.api+json
+Accept: application/vnd.api+json
+```
+
+```json
+{
+  "data": [
+    { "type": "io.cozy.files", "id": "417c4e58-e2e4-11e6-b7dc-2b68ed7b77f4" },
+    { "type": "io.cozy.files", "id": "4504f55c-e2e4-11e6-88f2-d77aeecab549" },
+    { "type": "io.cozy.files", "id": "4587727a-e2e4-11e6-bfe9-ef1be7df7f26" },
+    { "type": "io.cozy.files", "id": "45d591d0-e2e4-11e6-ab9a-ff3b218e31cc" }
+  ]
+}
+```
+
+#### Response
+
+```http
+HTTP/1.1 204 No Content
+Content-Type: application/vnd.api+json
+```
+
+### DELETE /files/:type/:doc-id/relationships/references
+
+This bulk deletion of refences on many files can be useful when an album or
+playlist is deleted.
+
+#### Request
+
+```http
+DELETE /files/io.cozy.playlists/e9308dc2-e2e3-11e6-b685-fb88662613d4/relationships/references HTTP/1.1
+Content-Type: application/vnd.api+json
+Accept: application/vnd.api+json
+```
+
+```json
+{
+  "data": [
+    { "type": "io.cozy.files", "id": "417c4e58-e2e4-11e6-b7dc-2b68ed7b77f4" },
+    { "type": "io.cozy.files", "id": "4504f55c-e2e4-11e6-88f2-d77aeecab549" },
+    { "type": "io.cozy.files", "id": "4587727a-e2e4-11e6-bfe9-ef1be7df7f26" },
+    { "type": "io.cozy.files", "id": "45d591d0-e2e4-11e6-ab9a-ff3b218e31cc" }
+  ]
+}
+```
+
+#### Response
+
+```http
+HTTP/1.1 204 No Content
+Content-Type: application/vnd.api+json
+```
+
+## Usage
+
+### Modification of a referenced file
+
+Before an application updates a file, it can check if the file has some
+references. If it is the case, it may offer to the user two choices:
+
+- update the file with the new version (the albums and playlists will use the
+  new version)
+- save the new version as a new file and preserve the old file (the old file
+  may be moved to a `originals` directory).
```

Comment:
So, the idea is that we want that kind of things but we haven't any clue of what to do it technically. My input here would be that you can organize the folders/original files like you want, as long as we can do what we want with the related permissions: the app Photos should be allowed to access the files it can restores, the app Music too, but they shouldn't be allowed to mess around with files they aren't related to. What do you think, is it something to keep in mind?